### PR TITLE
[MRG] Add testing for Nilearn setup & install & Fix the broken dependency installation

### LINKED ISSUE #1993: problem with setup.py: some dependencies are not installed by pip
pip install . in a fresh environment does not install dependencies

### LINKED ISSUE #2199: Testing Nilearn setup, build, & installation
**What would you like changed/added and why?**
Use AzurePipelines unused additional CI jobs to test Nilearn installation.

**What would be the benefit?**
It will detect and forestall errors during building and installing Niearn preventing the need for PR like #2198 & help detect issues like #1993  . The additional time required for this would be very low.

### PR DESCRIPTION
**Pull request by kchawla-pi**
Fixes #1993 , Fixes #2199

### CODE REVIEW

**Comment by kchawla-pi** [comment]
@jeromedockes @GaelVaroquaux 
This works on all except Appveyor. So far I have had to choose between passing on Appveyor and passing on the other CIs. We will be removing Appveyor soon.

What do you suggest? How do I fix this, or should I remove Appveyor now?

File: setup.py [hunk 1]
```diff
@@ -51,11 +66,6 @@ def is_installing():
         module_check_fn = _VERSION_GLOBALS['_check_module_dependencies']
         module_check_fn(is_nilearn_installing=True)
 
-    install_requires = \
-        ['%s>=%s' % (mod, meta['min_version'])
-            for mod, meta in _VERSION_GLOBALS['REQUIRED_MODULE_METADATA']
-            if not meta['required_at_installation']]
```
Comment (kchawla-pi) [inline setup.py]: @GaelVaroquaux 
This line was filtering out packages which were marked as required: True in `version.py`. I don't understand why this is here at all, and it concerns me that it is so much work to understand this.
Can you tell me why this line is here?
We document &/or streamline the setup files.
cc: @jeromedockes

**Comment by kchawla-pi** [comment]
AZP's `test_installing_on_windows` is failing because the file path spec is different from unix shells. I'll work on that from home on my Windows machine.

**Comment by kchawla-pi** [comment]
The drop in reported code coverage is consistent but misleading. Coverage has not changed.

**Comment by kchawla-pi** [comment]
Ready for review.

**Comment by kchawla-pi** [comment]
@jeromedockes @GaelVaroquaux 
You can skip the files related to the CI config, is that's unfamiliar. I am primarily interested in the changes I made to Python files, esp `setup.py`.

**Comment by bthirion** [comment]
LGTM.

**Pull request merged**